### DISCUSSION TITLE
Best practices in SvelteKit for i18n / L10n / internationalization / localization?

### DISCUSSION THREAD

**Discussion by rosano** [sveltejs/kit#9618] [Q&A]
I've seen the issues for [brainstorming](https://github.com/sveltejs/kit/issues/553) and [translations](https://github.com/sveltejs/kit/issues/1274), and would like to know: is there a recommended approach until the built-in solution is released?

Some solutions I've seen include:
- [svelte-i18n usage with SvelteKit](https://github.com/kaisermann/svelte-i18n/blob/main/docs/Svelte-Kit.md)
- [typesafe-i18n integration with SvelteKit](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/adapter-svelte)
- [implementation for sveltevietnam.dev](https://github.com/sveltevietnam/sveltevietnam.dev/issues/2)

**Comment by anthumchris** [comment]
I tried researching this today and couldn't find international language support in the [SvelteKit FAQ](https://kit.svelte.dev/docs/faq) nor search results below:
- https://kit.svelte.dev/search?q=i11n
- https://kit.svelte.dev/search?q=internationalization
- https://kit.svelte.dev/search?q=language
- https://kit.svelte.dev/search?q=locale

Related:
- https://github.com/sveltejs/kit/issues/12160
- https://github.com/sveltejs/kit/discussions/12001 mentions a i18n limitation, but no implementation method was detailed.

**Reply by mobilars** [reply]
Would be great to know if there is some official support for multiple languages planned, so that we don't go for one solution and then Sveltekit comes with a new solution that's better long term.

**Comment by vnphanquang** [comment]
Maintainer of sveltevietnam.dev here. Just want to say that over the years, our i18n implementation has gone through some major iterations, the latest of which now leverages [Async Svelte](https://svelte.dev/docs/svelte/await-expressions) and [Remote Function](https://svelte.dev/docs/kit/remote-functions) from SvelteKit. 

The solution is packaged as [@sveltevietnam/i18n](https://github.com/sveltevietnam/sveltevietnam.dev/tree/main/packages/i18n). A complementary blog post about it is at "[Internationalisation with Remote Function — A Proof of Concept](https://www.sveltevietnam.dev/en/blog/20251105-internationalisation-with-remote-function-a-proof-of-concept)".

There is also [wuchale](https://wuchale.dev/) and [paraglide](https://inlang.com/m/dxnzrydw/paraglide-sveltekit-i18n), each with its own merits and gotchas.

I do also believe that first party support should be realised at some point in the future.
